lib/driver.rb

```diff
@@ -42,6 +43,34 @@ def add_trip(trip)
       end
 
       @trips << trip
+    end # add_trip
+
+    def total_revenue
+      tax = 1.65
+      driver_pay = 0.8
+      gross_pay = 0
+
+      trips.each do |trip|
+        gross_pay += trip.cost - tax
+      end
+      total_revenue = gross_pay * driver_pay
```

Comment:
This method will hit an error if you try to run it on a driver with an incomplete trip. There are a couple of workarounds for this.

You could explicitly ignore trips with a `nil` cost:

```ruby
trips.each do |trip|
  if trip.cost.nil?
    next
  end
  # ... add to the total ...
end
```

Or even better, you could write a helper method that returns a list of complete trips:

```ruby
def completed_trips
  return @trips.reject{ |t| t.end_time.nil? }
end

def total_revenue
  completed_trips.each do |trip|
    # ... same logic as before ...
  end
end
```